Реализация работы с трекером через http

nishka322 opened the pull request.

Commit:
Реализация работы с трекером через http

VladimirTitov4 commented on test/http/HttpTaskManagerHistoryTest.java:
```diff
@@ -0,0 +1,102 @@
+        HttpResponse<String> response2 = client.send(request2, HttpResponse.BodyHandlers.ofString());
+        assertEquals(201, response2.statusCode(), "Неверный статус код для второй задачи");
+
+        Thread.sleep(100);
```
А для чего здесь засыпание потока ? Не самый лучший подход, да и в целом засыпать программа не должна

Commit:
Исправления по ревью.

1. Добавлен новый enum для запросов.
2. Изменена обработка запросов в хэндлерах.
3. Исправлен тест testGetHistory(): вместо ожидания добавления через sleep() получаем явные ID через сервер, удалил вспомогательные методы.
